### DISCUSSION TITLE
Feature Request: ignore file comment

### DISCUSSION THREAD

**Discussion by GabenGar** [rome/tools#4070] [Suggestions]
I run a codegen which creates a bunch of files and some of the code in them is generated by a 3rd-party package. This triggers a lot of Rome rules and I'd rather not let it mess with the resulting files.
I assumed I could add a comment to each file akin to `@ts-nocheck`, but I [didn't find](https://docs.rome.tools/linter/#ignoring-code) a way to ignore a file from a comment in the docs, so it looks like there is no way as for now.
Hence feature request.

**Comment by ematipico** [comment]
Your request is to have a comment to disable **completely the linter** for a certain file?

If so, what's preventing you from using `linter.ignore` via configuration file? https://docs.rome.tools/configuration/#linterignore

I think it's important to understand the use case

**Reply by GabenGar** [reply]
> Your request is to have a comment to disable completely the linter for a certain file?

Linter and/or formatter I guess. 
The files in question are outside of the Rome's workspace. It's the VSCode extension which cries about errors. Although I eventually plan to use Rome for all workspaces and that will become a bigger issue.
The usecase is exactly that - being able to to ignore files without needing to change the config (which can get pretty complicated once you need to ignore files selectively within nested subfolders). And I also have an apathy towards string literal globs/paths after dealing with them in multi-platform contexts.

**Comment by jpike88** [comment]
I would like to have this as a possible way to ignore a file. It suits conventions set by typescript, eslint etc, and allows for a more flexible way to ignore certain files without setting their paths in stone in a config file somewhere, something more prone to accidental/undefined behaviour if the files are moved to a different path etc...

**Reply by ematipico** [reply]
Personally, I am quite skeptical about adding a feature like this in Rome's linter, for different reasons:

- once enabled, lint rules should be followed in the code base. It makes sense to disable them in some places (comment suppression) where the user can add a **security comments** explaining why it's safe to disable a rule;
- disabling the linter for a file can be achieved via configuration. Via configuration, if a file is moved, the user is reminded of that. Using a comment, the user can forget that the linter is disabled for that file which could cause unpleasant issues;

**Reply by jpike88** [reply]
I agree with that point of view as well, my solution is just to ensure everything impresses the linter, which I managed to do today without much added effort.
